Update install instructions to Python 3

rgaiacs opened the pull request:
**Only merge this after https://github.com/swcarpentry/python-novice-inflammation/pull/155.**

rgaiacs committed:
Update install instructions to Python 3

gvwilson commented on index.html:
```diff
@@ -491,37 +490,29 @@ <h4 id="editor-linux">Linux</h4>
-      </ul>
+      <p>
+        Download the <a href="http://continuum.io/downloads#py34">Python 3 installer</a>
+        <a href="https://store.continuum.io/cshop/anaconda/">Anaconda</a>.
```
I know what you mean, but the wording is a bit awkward - odds are good someone will click on the Anaconda link thinking that they need to do that _as well as_ installing Python 3.  How about just saying "Download _this_ Python 3 installer (which comes with all the core scientific libraries)." and skip the mention of Anaconda.

gvwilson commented on index.html:
```diff
@@ -491,37 +490,29 @@ <h4 id="editor-linux">Linux</h4>
-        <a href="http://continuum.io/downloads.html">Anaconda</a>.
+        We recommend the
+        <a href="http://continuum.io/downloads#py34">all-in-one scientific Python 3 installer</a>
+        <a href="https://store.continuum.io/cshop/anaconda/">Anaconda</a>.
```
As above.

gvwilson commented on index.html:
```diff
@@ -530,7 +521,7 @@ <h4 id="python-linux">Linux</h4>
           Download the installer that matches your operating
           system and save it in your home folder.
-          Download the default Python 2 installer (do not follow the link to version 3).
+          Download the Python 3 installer.
```
Do you need to say "Do not follow the link to Python 2" ?

rgaiacs replied:
I think that saying it will be confusing to students.

rgaiacs commented:
http://continuum.io/downloads#py34 isn't working as expected. I'm waiting reply from Continuum.

rgaiacs committed:
Update installation test scripts

rgaiacs commented:
@gvwilson
1. Could you check the new wording?
2. I changed the minimum version of Python at the testing script to 3.3
3. I dropped the compatibility with Python 2 and < 3.3. Are you OK with it? If yes, I want to clean a little more the testing script. If no, I will drop 51781bf.
4. I need some time to run the testing script in a Windows virtual machine (can only do it at the weekend).

gvwilson commented on index.html:
```diff
@@ -491,37 +492,26 @@ <h4 id="editor-linux">Linux</h4>
-        </li>
-      </ul>
+      <p>
+        Download <a href="http://continuum.io/downloads#py34">this Python 3 installer</a>
```
Is this link now working?

rgaiacs replied:
The link always goes to Python 2. You need Javascript to get the link for Python 3.

gvwilson replied:
Ouch.  That's really bad news :-(  Have you spoken to Continuum about 
fixing it?

rgaiacs replied:
> Have you spoken to Continuum about fixing it?

The first thing that I did. Waiting they reply.

gvwilson replied:
Gotcha - gotta say, that seems like a bug to me...

gvwilson commented on index.html:
```diff
@@ -491,37 +492,26 @@ <h4 id="editor-linux">Linux</h4>
-        <a href="http://continuum.io/downloads.html">Anaconda</a>.
+        Download <a href="http://continuum.io/downloads#py34">this Python 3 installer</a>
+        and install it.
         (Installation requires using the shell and if you aren't
```
"Installation requires using the shell.  If you aren't..."

gvwilson commented on setup/swc-installation-test-2.py:
```diff
@@ -22,23 +22,9 @@
 import distutils.ccompiler as _distutils_ccompiler
 import fnmatch as _fnmatch
-try:  # Python 2.7 and 3.x
-    import importlib as _importlib
```
Hm - we should still _run_ when it's Python 2.7, just tell people "your Python is out of date".

gvwilson commented:
> Could you check the new wording?
> Done
> 1. I changed the minimum version of Python at the testing script to 3.3
> 
> +1
> 1. I dropped the compatibility with Python 2 and < 3.3. Are you OK
>    with it? If yes, I want to clean a little more the testing script.
>    If no, I will drop 51781bf
>    https://github.com/swcarpentry/workshop-template/commit/51781bfc240f9c00efb02218b6414b2a40629886.
> 
> We should keep the compatibility - it should run, but say "you're out of 
> date".
> 1. I need some time to run the testing script in a Windows virtual
>    machine (can only do it at the weekend).
> 
> ...or ask on 'discuss' for help to do it sooner?

Thanks,
Greg

rgaiacs commented:
@gvwilson 
1.  Could you check the new wording? Unfortunately, learners will need to click on "I want Python 3.X"
2.  I restored the compatibility with Python 2 as requested.

If the new wording is OK, I think this is ready to merge.

gvwilson commented on index.html:
```diff
@@ -493,47 +494,33 @@ <h4 id="editor-linux">Linux</h4>
+        <li>Open <a href="http://continuum.io/downloads">http://continuum.io/downloads</a> with your web browser.</li>
+        <li>Click on "I want Python 3.X" link.</li>
+        <li>Download this Python 3 installer.</li>
+        <li>Install Python 3 using all of the defaults for installation.</li>
```
Why don't they have to make this default the way they do for Linux?

gvwilson commented:
One comment, but other than that, looks good to merge.

rgaiacs commented:
@gvwilson Can we merge this and have another PR when Continuum Analytics fixed their site?

wking commented:
This slipped through without me noticing it, but it's just come up in
#312.  I don't think we want to bump the minimum Python version in
setup/swc-installation-test-1.py like 88c78b8 does.  That's just the
minimum version required to _parse_ swc-installation-test-2.py [1](https://github.com/wking/swc-setup-installation-test/#for-instructors).
If a particular lesson/workshop needs a higher version of Python, they
should be bumping this [2](https://github.com/wking/swc-setup-installation-test/blob/74844ea43e4df102d9013eae92d19f7f924ee00b/swc-installation-test-2.py#L665).  Alternatively, we could land
wking/swc-setup-installation-test#9 and the lesson/workshop could
declare a dependency on:

  "python": {
    "minimum-version": [3, 3].
  }
